src/NuGet.Indexing/Extraction/CatalogPackageReader.cs

```diff
@@ -46,6 +46,23 @@ public override IEnumerable<string> GetFiles()
             }
         }
 
+        public override IReadOnlyList<PackageType> GetPackageTypes()
+        {
+            var packageTypes = new List<PackageType>();
+            var array = _catalogItem.GetJArray("packageTypes");
+            if (array == null)
+            {
+                return packageTypes;
+            }
+
+            foreach (var entry in array)
+            {
+                packageTypes.Add(new PackageType((string)entry["type"], new Version((string)entry["version"])));
```

Comment:
Version should always be defined. It will either be defined by the package metadata, or filled in with a default value.